### ISSUE TITLE
HyperV: Vagrant Up Failure after Vagrant Destroy

### ISSUE THREAD

**Issue by darkn3rd** [hashicorp/vagrant#10457]
Cannot Bring up System with `vagrant up --provider=hyperv` after `vagrant destroy --force`

### Vagrant version
Run `vagrant -v` to show the version. If you are not running the latest version
of Vagrant, please upgrade before submitting an issue.

### Host operating system
* Windows 10 Pro, 64-bit
### Guest operating system

* Ubuntu 16.04 Xenial

### Vagrantfile
```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "bento/ubuntu-16.04"
end
```

### Debug output

https://gist.github.com/darkn3rd/e5a49f49feee5c5d412c1d51526e1f51

### Expected behavior

Vagrant up works.  Whatever was there was cleaned up.  Docs or output other than stacktrace, on suggestions to fix.

### Actual behavior
```
C:/HashiCorp/Vagrant/embedded/mingw64/lib/ruby/2.4.0/fileutils.rb:1335:in `rmdir': Directory not empty @ dir_s_rmdir - C:/Users/joaquin/virtualenv_test/.vagrant/machines/default/hyperv/Virtual Hard Disks (Errno::ENOTEMPTY)
        from C:/HashiCorp/Vagrant/embedded/mingw64/lib/ruby/2.4.0/fileutils.rb:1335:in `block in remove_dir1'
        from C:/HashiCorp/Vagrant/embedded/mingw64/lib/ruby/2.4.0/fileutils.rb:1349:in `platform_support'
        from C:/HashiCorp/Vagrant/embedded/mingw64/lib/ruby/2.4.0/fileutils.rb:1334:in `remove_dir1'
        from C:/HashiCorp/Vagrant/embedded/mingw64/lib/ruby/2.4.0/fileutils.rb:1327:in `remove'
        from C:/HashiCorp/Vagrant/embedded/mingw64/lib/ruby/2.4.0/fileutils.rb:689:in `block in remove_entry'
        from C:/HashiCorp/Vagrant/embedded/mingw64/lib/ruby/2.4.0/fileutils.rb:1384:in `ensure in postorder_traverse'
        from C:/HashiCorp/Vagrant/embedded/mingw64/lib/ruby/2.4.0/fileutils.rb:1384:in `postorder_traverse'
        from C:/HashiCorp/Vagrant/embedded/mingw64/lib/ruby/2.4.0/fileutils.rb:687:in `remove_entry'
        from C:/HashiCorp/Vagrant/embedded/mingw64/lib/ruby/2.4.0/fileutils.rb:555:in `block in rm_r'
        from C:/HashiCorp/Vagrant/embedded/mingw64/lib/ruby/2.4.0/fileutils.rb:551:in `each'
        from C:/HashiCorp/Vagrant/embedded/mingw64/lib/ruby/2.4.0/fileutils.rb:551:in `rm_r'
        from C:/HashiCorp/Vagrant/embedded/mingw64/lib/ruby/2.4.0/pathname.rb:587:in `rmtree'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.0/gems/vagrant-2.2.0/lib/vagrant/machine.rb:360:in `block in id='
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.0/gems/vagrant-2.2.0/lib/vagrant/machine.rb:358:in `each'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.0/gems/vagrant-2.2.0/lib/vagrant/machine.rb:358:in `id='
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.0/gems/vagrant-2.2.0/lib/vagrant/machine.rb:147:in `initialize'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.0/gems/vagrant-2.2.0/lib/vagrant/vagrantfile.rb:81:in `new'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.0/gems/vagrant-2.2.0/lib/vagrant/vagrantfile.rb:81:in `machine'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.0/gems/vagrant-2.2.0/lib/vagrant/environment.rb:716:in `machine'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.0/gems/vagrant-2.2.0/lib/vagrant/plugin/v2/command.rb:180:in `block in with_target_vms'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.0/gems/vagrant-2.2.0/lib/vagrant/plugin/v2/command.rb:204:in `block in with_target_vms'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.0/gems/vagrant-2.2.0/lib/vagrant/plugin/v2/command.rb:186:in `each'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.0/gems/vagrant-2.2.0/lib/vagrant/plugin/v2/command.rb:186:in `with_target_vms'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.0/gems/vagrant-2.2.0/plugins/commands/up/command.rb:87:in `execute'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.0/gems/vagrant-2.2.0/lib/vagrant/cli.rb:54:in `execute'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.0/gems/vagrant-2.2.0/lib/vagrant/environment.rb:291:in `cli'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.0/gems/vagrant-2.2.0/bin/vagrant:164:in `<main>'
```

### Steps to reproduce
1. `vagrant destroy --force`
1. `vagrant up --provider=hyperv`

**Comment by darkn3rd** [comment]
Alright, I think there was a lock on the resource (guessing), and Vagrant is unable to detect a lock, so destroy fails, but returns a success.  Then afterward, destroy thinks everything is destroyed, but it really isn't destroy.  If there is a lock, destroy needs to just fail (short-circuit) and let users know there is a lock.

**Comment by briancain** [comment]
@darkn3rd - Can you try upgrading to the latest version of Vagrant to see if this has been fixed? I haven't been able to reproduce it with the latest changes on my end.

**Comment by chrisroberts** [comment]
This should be fixed via #10406 which was included in the 2.2.1 release. If you still experience this behavior after upgrading, please open a new issue and I will investigate further.

Thanks!
